src/app/containers/Labels/modals/Edit.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import { Modal, HeaderModal, FooterModal, ContentModal, Button } from 'react-components';
+import { LABEL_TYPES } from 'proton-shared/lib/constants';
+
+import NewLabelForm from '../../../components/Labels/NewLabelForm';
+
+function EditLabelModal({ show, label, onSubmit, ...props }) {
+    const [state, setState] = useState(label);
+    const handleSubmit = () => onSubmit(state);
+    const handleUpdate = setState;
+
+    const I18N = {
+        edition({ Name, Exclusive } = {}) {
+            if (Exclusive === LABEL_TYPES.LABEL) {
+                return c('Label/folder modal').t`Edit label: ${Name}`;
+            }
+            return c('Label/folder modal').t`Edit folder: ${Name}`;
+        },
+        create(label, type) {
+            if (type === 'label') {
+                return c('Label/folder modal').t('Create a new label');
+            }
+            return c('Label/folder modal').t('Create a new folder');
+        }
+    };
+
+    return (
+        <Modal show={show} {...props}>
```

Comment:
I prefer if you not do `{...props}`, this can be dangerous if we change `<Modal>` and add an attribute sharing the same name.